Tests/TestingTests/Support/FileHandleTests.swift

```diff
@@ -0,0 +1,146 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+@testable import Testing
+private import TestingInternals
+
+#if !SWT_NO_FILE_IO
+// NOTE: we don't run these tests on iOS (etc.) because processes on those
+// platforms are sandboxed and do not have arbitrary filesystem access.
+#if os(macOS) || os(Linux) || os(Windows)
+@Suite("FileHandle Tests")
+struct FileHandleTests {
+  @Test("Can get file descriptor")
+  func fileDescriptor() throws {
+    #if os(Windows)
+    let tmpFile: SWT_FILEHandle = try {
+      var file: SWT_FILEHandle?
+      try #require(0 == tmpfile_s(&file))
+      return file!
+    }()
+    #else
+    let tmpFile = try #require(tmpfile())
+    #endif
+    let fileHandle = FileHandle(unsafeCFILEHandle: tmpFile, closeWhenDone: true)
+    try fileHandle.withUnsafePOSIXFileDescriptor { fd in
+      try #require(fd != nil)
+    }
+  }
+
+#if os(Windows)
+  @Test("Can get Windows file HANDLE")
+  func fileHANDLE() throws {
+    let tmpFile: SWT_FILEHandle = try {
+      var file: SWT_FILEHandle?
+      try #require(0 == tmpfile_s(&file))
+      return file!
+    }()
+    let fileHandle = FileHandle(unsafeCFILEHandle: tmpFile, closeWhenDone: true)
+    try fileHandle.withUnsafeWindowsHANDLE { handle in
+      try #require(handle != nil)
+    }
+  }
+#endif
+
+  @Test("Can write to a file")
+  func canWrite() throws {
+    // NOTE: we are not trying to test mkstemp() here. We are trying to test the
+    // capacity of FileHandle to open a file for writing, and need a temporary
+    // file to write to.
+    let fileName = "can_write_to_file_\(UInt64.random(in: 0 ..< .max))"
+#if os(Windows)
+    let cPath = try #require(_tempnam("c:\\tmp", fileName))
```

Comment:
This path is not really guaranteed to exist. Computing a temporary path is pretty complicated, so might be better to use `GetTempPathW` here.

Reply:
This code is borrowed directly from Microsoft's documentation, but I can change it to just use `tmpnam_s()` instead.